frontend/src/metabase/components/DriverWarning/DriverWarning.tsx

```diff
@@ -29,34 +35,120 @@ const DriverWarning = ({
   const oldEngine = oldEngineKey ? engines[oldEngineKey] : undefined;
   const handleChangeToOld = () => oldEngineKey && onChange?.(oldEngineKey);
 
+  const warnings = [];
+
   if (newEngine) {
-    return (
-      <WarningRoot hasBorder={hasBorder}>
-        {t`This driver will be removed in a future release.`}{" "}
-        {jt`We recommend you upgrade to the ${(
-          <WarningLink key="link" onClick={handleChangeToNew}>
-            {t`new ${newEngineName} driver`}
-          </WarningLink>
-        )}.`}
-      </WarningRoot>
+    warnings.push(
+      <NewEngineWarning
+        key="new"
+        engineName={newEngineName || ""}
+        handleChange={handleChangeToNew}
+      />,
     );
   }
 
   if (oldEngine) {
-    return (
-      <WarningRoot>
-        {t`This is our new ${engineName} driver.`}{" "}
-        {t`The old driver has been deprecated and will be removed in a future release.`}{" "}
-        {jt`If you really need to use it, you can ${(
-          <WarningLink key="link" onClick={handleChangeToOld}>
-            {t`find it here`}
-          </WarningLink>
-        )}.`}
-      </WarningRoot>
+    warnings.push(
+      <OldEngineWarning
+        key="old"
+        engineName={engineName}
+        handleChange={handleChangeToOld}
+      />,
+    );
+  }
+
+  if (engineSourceType === "community") {
+    warnings.push(<CommunityDriverWarning key="community" />);
+  } else if (engineSourceType === "partner") {
+    warnings.push(
+      <PartnerDriverWarning
+        key="partner"
+        sourceName={engine?.source?.contact?.name}
+        sourceContact={engine?.source?.contact?.address}
```

Comment:
I'm curious, does this mean that if engine.source.contact is defined, that you should expect an address?

Reply:
if there's a name but no address in contact, it'll fall back gracefully to just showing the name